File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        r = n
        res = 0
        for i in range(m-1, -1, -1):
            for j in range(r):
                if grid[i][n-1-j] < 0:
                    res += 1
                else:
                    r = j
                    break
            if not(r):
                break
        return res

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.rs
impl Solution {
    pub fn count_negatives(grid: Vec<Vec<i32>>) -> i32 {
        let m: i32 = grid.len() as i32;
        let n: i32 = grid[0].len() as i32;
        let mut r: i32 = n;
        let mut res: i32 = 0;
        for i in (0..m).rev() {
            for j in 0..r {
                if (grid[i as usize][(n-1-j) as usize] < 0) {
                    res += 1;
                }
                else {
                    r = j;
                    break;
                }
            }
            if (r == 0) {
                break;
            }
        }
        return res;
    }
}
